com.ibm.wala.core/src/com/ibm/wala/types/Selector.java

```diff
@@ -24,8 +27,17 @@
 
   private final Descriptor descriptor;
   
+  private final String s;
+  
+  private static final Map<String,Selector> CACHE = new HashMap<String,Selector>();
```

Comment:
Few comments:
* Can you explain the reason for this change?  Was creation of `Selector` objects causing a memory issue, or running time?  The `CACHE` field introduces a memory leak, and we have these in other places in the code, but I'd like to understand why it's needed now.
* Rather than allocating `HashMap` directly, we try to use `HashMapFactory.make()` in `wala.core`.  This allocates a `LinkedHashMap` for better determinism and can help in debugging hashCode()/equals() problems.
* Maybe we should make the cache work for other languages too?